### PR TITLE
Adds keyboard shortcuts to PM UI Preview Window, License Acceptance and File Conflict dialog

### LINKED ISSUE #10691: File Conflict dialog does not have access keys
## Details about Problem

NuGet product used (NuGet.exe | Visual Studio | MSBuild.exe | dotnet.exe): Visual Studio

Product version:  16.9

Worked before? If so, with which NuGet version: 

## Repro steps and/or sample project

1. Create a .NET Framework ASP.NET WebForms project
2. Modify Scripts\bootstrap.js file
3. If bootstrap.js is not installed, install it via PM UI
4. Modify Scripts\bootstrap.js file, add a comment
5. Update bootstrap package via PM UI
6. Following dialog will appear

image

### Expected 
Buttons with Underscore: <ins>Y</ins>es, <ins>N</ins>o, Yes to <ins>A</ins>ll, No to A<ins>l</ins>l
### Actual
No Underscore: Yes, No, Yes to All, No to All

### Notes

This requires localization cycle

### PR DESCRIPTION
**Pull request by dominoFire**
## Bug

Fixes: https://github.com/NuGet/Home/issues/10691

Regression? Last working version: 

## Description

Accessibility improvement:
- Adds keyboard access keys to File Conflict Dialog, Preview Changes Window Dialog and License Acceptance Window by adding an underscore (_) to button texts. This improves usability with keyboard.
- Added support to access keys in Checkbox, disabled in CheckBox style before this change.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - ~[ ] Automated tests added~
  - ~[ ] Test exception~
  - [x] N/A: UI change, manually validated

- **Documentation**
  - ~[ ] Documentation PR or issue filled~
  - [x] N/A: Does not change main scenario features

- **Validation**

Manual validation: it shows the keyboard shortcuts when pressing 'Alt' key.

#### File Conflict Dialog

After:
image

Accessibility Insights test:
image

#### Preview Window Dialog

After:
image

Accessibility Insights test:
image

#### License Acceptance Window Dialog

Affter:
image

Accessibility Insights test:
image

### CODE REVIEW

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        /// Returns char next to the first underscore found
+        /// </summary>
+        /// <param name="value">A string resource</param>
+        /// <param name="targetType">No use</param>
+        /// <param name="parameter">No use</param>
+        /// <param name="culture">No use</param>
+        /// <returns></returns>
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value == null)
```
Comment (jebriede) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs]: Lines 22 through 32 could be replaced with:

```
var input = value as string;
if (input == null)
{
   return string.Empty;
}
```

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx [hunk 2]
```diff
@@ -265,16 +265,20 @@
     <value>File Conflict</value>
   </data>
   <data name="Button_No" xml:space="preserve">
-    <value>No</value>
+    <value>_No</value>
+    <comment>Button_Yes, Button_No, ButtonYesToAll, Button_NoToAll should have underscore in different letters. See https://docs.microsoft.com/en-us/windows/uwp/design/input/access-keys#choose-access-keys</comment>
```
Comment (jebriede) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: ButtonYesToAll should be Button_YesToAll in all the comments.

File: test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Converters/AcceleratorKeyConverterTests.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+using System;
+using System.Globalization;
+using Moq;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test.Converters
+{
+    public class AcceleratorKeyConverterTests
+    {
+        [Theory]
```
Comment (jebriede) [inline test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/Converters/AcceleratorKeyConverterTests.cs]: Another edge case I'd add is when the underscore is at the end of the string. This will ensure any changes to the algorithm preserve the behavior. Example: "Yes_".

**Review by jebriede — DISMISSED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs [hunk 4]
```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Windows.Data;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class AcceleratorKeyConverter : IValueConverter
```
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs]: Is this something you copied from another repo? I did a quick search and couldn't find, so wondering if there's a common practice that avoids converters?

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx (hunk 2 above)
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: I'd suggest checking with localization team to make sure they understand this, if you haven't already.

**Review by donnie-msft — COMMENTED** [review]
Did you run Accessibility Insights before/after these changes? I don't think it's a hard violation, but there may have been a warning that your PR resolves. 

Also, I think it's just good to make sure AI reflects the new change.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs (hunk 4 above)
Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs]: nit: Also, I believe these are actually "Access Keys", not accelerators.

**Comment by donnie-msft** [comment]
I briefly played with a few options, and I'm pretty confused as to how WPF is handling these.
Setting the Automation Property of `AccessKey` or `AcceleratorKey` did nothing by itself. Only if I set button's text with `Back_ground` does it actually affect the behavior :D
image

My observation:

- "_" in content 
  - makes the shortcut key work 
  - sets the AccessKey property in AI
  
- "AutomationProperty.AccessKey" 
  - property **does not** reflect in AI
  - does not enable shortcut
  - if set with "_" content, enables shortcut BUT breaks NVDA/Narrator announcement
  
- "AutomationProperty.AcceleratorKey" 
  - property does reflect in AI
  - no behavior
  - no NVDA/Narrator announcement
  - if set with "_" content
    -  NVDA announces the key twice (assuming content + AcceleratorKey property)
    -  Narrator as expected
    - AI reflects both!
    
image

**Comment by dominoFire** [comment]
Need to verify keyboard accelerators in another languages. For example, in Russsian language, it does not show keyboard accelerators.

image

**Comment by dominoFire** [comment]
Reopening to finish work.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx (hunk 2 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: Will follow up with them once this PR is merged.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs (hunk 4 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/AcceleratorKeyConverter.cs]: Thanks for poiting out. The correct property should be AccessKey

It seems setting AutomationProperties.AccessKey has no effect on keyboard accelertor behaviour.

So, I'm going to remove converter.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx [hunk 5]
```diff
@@ -260,21 +260,25 @@
...
     <value>Preview</value>
-  </data>
+  </data>`
   <data name="WindowTitle_FileConflict" xml:space="preserve">
     <value>File Conflict</value>
   </data>
   <data name="Button_No" xml:space="preserve">
-    <value>No</value>
+    <value>_No</value>
+    <comment>Button_Yes, Button_No, Button_YesToAll, Button_NoToAll should have underscore in different letters. See https://docs.microsoft.com/windows/uwp/design/input/access-keys#choose-access-keys</comment>
```
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: nit: this URL redirects to `https://docs.microsoft.com/en-us/windows/apps/design/input/access-keys#choose-access-keys`

**Review by donnie-msft — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx (hunk 5 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: That's correct, because MSDN docs redirect to the translated page in the language specified in the client request. It's a good practice to remove language path in MSDN docs URIs to let the user decide language.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj [hunk 6]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj]: Pleasa consider reverting this file since not actual change.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/FileConflictDialog.xaml [hunk 7]
```diff
@@ -75,4 +75,4 @@
       VerticalAlignment="Top"
       AutomationProperties.Name="{Binding Text}"/>
   </DockPanel>
-</ui:VsDialogWindow>
```
Comment (erdembayar) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/FileConflictDialog.xaml]: This one too, better to revert.

**Review by erdembayar — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj (hunk 6 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/NuGet.PackageManagement.UI.csproj]: reverted

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/FileConflictDialog.xaml (hunk 7 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/FileConflictDialog.xaml]: reverted

**Review by jeffkl — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx (hunk 5 above)
Comment (donnie-msft) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: No, I'm pointing out that it redirected from a UWP specific url to a generic Windows url.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources/Resources.xaml [hunk 8]
```diff
@@ -213,6 +213,7 @@
               Grid.Column="1"
               Focusable="False"
               Margin="{TemplateBinding Padding}"
+              RecognizesAccessKey="True"
```
Comment (donnie-msft) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Resources/Resources.xaml:216]: 👀

Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Resources/Resources.xaml:216]: Added to support access keys in Checkbox, disabled in this style before this change.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx (hunk 5 above)
Comment (dominoFire) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Resources.resx]: Updated URL's
